src/common/io.py

```diff
@@ -231,172 +231,192 @@ def _find_underlying_state(self, name):
             return ".".join(spl)
         else:
             return name
-    
+
     def _create_derivative_from_state(self, name):
         """
         Create a derivative variable from a state by adding for instance
         to PI.x -> der(PI.x).
         """
         return "der("+name+")"
-    
+
     def _convert_dx_name(self, name):
         """
-        Internal method for converting the derivative variable into the 
-        "other" convention. A derivative variable can either be on the 
+        Internal method for converting the derivative variable into the
+        "other" convention. A derivative variable can either be on the
         form PI.der(x) and der(PI.x).
-        
+
         Returns the original name if the name was not a derivative name.
         """
         spl = name.split(".") #Split name
-        
+
         if spl[0].startswith("der("): #der(PI.x)
             spl[0] = spl[0][4:] #Remove der
             spl[-1] = "der("+spl[-1] #Add der
-            
+
             return ".".join(spl) #PI.der(x)
 
         elif spl[-1].startswith("der("): #PI.der(x)
             spl[0] = "der("+spl[0] #Add der
             spl[-1] = spl[-1][4:] #Remove der
-            
+
             return ".".join(spl)
 
         else: #Variable was not a derivative variable
             return name
 
 class ResultCSVTextual:
+    """ Class representing a simulation or optimization result loaded from a CSV-file. """
     def __init__(self, filename, delimiter=";"):
-        
-        fid = codecs.open(filename,'r','utf-8')
-        
+        """
+        Load a result file written on CSV format.
+
+        Parameters::
+
+            filename --
+                Name of file or stream object which the result is written to.
+                If filename is a stream object, it needs to support 'readline' and 'seek'.
+
+            delimiter --
+                The delimiter the data values is separated by.
+                Default: ";""
+        """
+
+        if isinstance(filename, str):
+            fid = codecs.open(filename,'r','utf-8')
+        else: #assume stream
+            if not(hasattr(filename, 'readline') and hasattr(filename, 'seek')):
+                raise JIOError("Given stream needs to support 'readline' and 'seek' in order to retrieve the results.")
+            fid = filename
+            fid.seek(0, 0) # need to start reading from beginning
```

Comment:
Can we assume that we start from the beginning or do we need this?

Reply:
It depends on what kind of user experience one wants. If we aren't at the beginning of the file we may get unexpected results back which will be hard for the user to trace.

Reply:
Ok